Sources/Basic/OutputByteStream.swift

```diff
@@ -405,3 +405,95 @@ public struct Format {
         return TransformedSeparatedListStreamable(items: items, transform: transform, separator: separator)
     }
 }
+
+/// Represents a stream which is backed to a file. Not for instantiating.
+public class FileOutputByteStream: OutputByteStream {
+
+    /// Closes the file flushing any buffered data.
+    public func close() {
+        // For subclasses to implement.
```

Comment:
Also, does it auto-close on `deinit`? Not sure what the right answer is, my preference might be that it assert in `deinit` if not closed.